earthly/mdlint/Earthfile

```diff
@@ -0,0 +1,61 @@
+VERSION 0.7
+
+MDLINT:
+    # Linting is done with MarkdownLint CLI2
+    # See: https://github.com/DavidAnson/markdownlint-cli2
+    # We use a special container which includes all rule extensions to markdownlint
+    # Notably, we rely on the `max-one-sentence-per-line` rule which helps keep
+    # diffs of markdown files small and legible.  This rule is only currently
+    # in the `next` branch of `markdownlint`.
+    # Container is built by:
+    #   https://github.com/DavidAnson/markdownlint-cli2/blob/next/docker/Dockerfile-rules
+    COMMAND
+
+    # Directory we need to run lint checks against.
+    ARG --required src
+
+    # Optional `fix` argument.  Must be either not set, or set to `--fix` anything else is an error.
+    ARG fix
+
+    # Unlikely this ever needs to be changed.
+    ARG cfg_file=.markdownlint-cli2.jsonc
+
+    # Ensure the `fix` argument only contains valid values.
+    RUN if [[ "${fix}" != "--fix" && "${fix}" != "" ]]; then \
+        echo "Invalid value for fix argument: ${fix}. It should be either --fix or not set." >&2; \
+        exit 1; \
+    fi
+
+    # Ensure the path to be checked has a mdlint config file.
+    RUN --no-cache if [[ ! -f $src/$cfg_file ]]; then \
+        (>&2 echo "$src/$cfg_file does not exist. Can not run mdLint."); \
+        exit 1; \
+    fi
+
+    # Status line for what we are about to do.
+    RUN --no-cache echo Linting Markdown Recursively from: $src
+
+    # Run the linter with the given arguments, and recursively check all markdown files.
+    # The directory to be checked `MUST` have a `.markdownlint-sli2.jsonc` file.
+    # cspell: words davidanson
+    WITH DOCKER --pull davidanson/markdownlint-cli2-rules:next
+        # Lint checks should not be cached.
+        RUN --no-cache \
+            docker run \
+            --rm \
+            -v $src:/workdir \
+            davidanson/markdownlint-cli2-rules:next \
+            "**/*.md" \
+            --config $cfg_file \
+            $fix
+    END
+
+# A Test and example invocation of the above UDC.
+mdlint-test:
+    # Test Markdown lint checks.
+    # Run with `earthly -P +mdlint-test
+    LOCALLY
+
+    ARG src=$(echo ${PWD}/../../)
```

Comment:
Does `${PWD}` resolve to the location of the Earthfile or to the location that `earthly` was invoked?

Reply:
My testing indicates it's the location of the Earthfile the target is in.
I use this kind of thing because earthly refuses to accept that `../../` is a valid path but `./` is.  It needs an absolute path if its not a child path of the Earthfile for some odd reason.

Reply:
This is one of my Earthly/Docker bugbears.  Any path that's valid should be valid.  If I can have a path relative to the Earthfile, I should be able to step up to a parent.  If its a security issue allow it with a CLI option.